lib/bundles/inspec-compliance/api/login.rb

```diff
@@ -0,0 +1,145 @@
+
+        return Login::ComplianceServer.login(options) if options['server_type'] == 'compliance'
+        Login::AutomateServer.login(options)
+      end
+
+      module AutomateServer
+        def self.login(options)
+          verify_thor_options(options)
+
+          options['url'] = options['server'] + '/compliance'
+          token = options['dctoken'] || options['token']
+          puts store_access_token(options, token)
+        end
+
+        def self.store_access_token(options, token)
+          token_info = if options['token']
+                         { type: 'usertoken', msg: 'automate user token' }
+                       else
+                         { type: 'dctoken', msg: 'data colletor token' }
+                       end
+
+          config = Compliance::Configuration.new
+
+          config.clean
+
+          config['automate'] = {}
+          config['automate']['ent'] = options['ent']
+          config['automate']['token_type'] = token_info[:type]
+          config['server'] = options['url']
+          config['user'] = options['user']
+          config['insecure'] = options['insecure'] || false
+          config['server_type'] = options['server_type']
+
+          config['token'] = token
+          config['version'] = Compliance::API.version(config)
+          config.store
+          "Stored configuration for Chef Automate: '#{config['server']}' with user: '#{config['user']}', ent: '#{config['automate']['ent']}' and your #{token_info[:msg]}"
+        end
+
+        # Automate login requires `--ent`, `--user`, and either `--token` or `--dctoken`
+        def self.verify_thor_options(o)
+          error_msg = []
+
+          error_msg.push('Please specify a server using `inspec compliance login https://SERVER`') if o['server'].nil?
+          error_msg.push('Please specify a user using `--user=\'USER\'`') if o['user'].nil?
+          error_msg.push('Please specify an enterprise using `--ent=\'automate\'`') if o['ent'].nil?
+
+          if o['token'].nil? && o['dctoken'].nil?
+            error_msg.push('Please specify a token using `--token=\'AUTOMATE_TOKEN\'` or `--dctoken=\'DATA_COLLECTOR_TOKEN\'`')
+          end
+
+          raise ArgumentError, error_msg.join("\n") unless error_msg.empty?
+        end
+      end
+
+      module ComplianceServer
+        def self.login(options)
+          compliance_verify_thor_options(options)
+
+          options['url'] = options['server'] + '/api'
+
+          if options['user'] && options['token']
+            success = true
+            msg = compliance_store_access_token(options, options['token'])
+          elsif options['user'] && options['password']
+            success, msg = compliance_login_user_pass(options)
+          elsif options['refresh_token']
+            success, msg = compliance_login_refresh_token(options)
+          end
+
+          raise msg unless success
+
+          puts msg
```

Comment:
Same feedback as above - I bet there's a good opportunity for reuse here, actually.